src/contracts/services/batchQuery.ts

```diff
@@ -64,27 +69,59 @@ const batchQuerySingleBatch$ = ({
   codeHash,
   queries,
   client,
+  minBlockHeightValidationOptions,
 }:{
   contractAddress: string,
   codeHash?: string,
   queries: BatchQueryParams[],
-  client: SecretNetworkClient
-}) => sendSecretClientContractQuery$({
-  queryMsg: msgBatchQuery(queries),
-  client,
-  contractAddress,
-  codeHash,
-}).pipe(
-  map((response) => parseBatchQuery(response as BatchQueryResponse)),
-  first(),
-  catchError((err) => {
-    if (err.message.includes('{wasm contract}')) {
-      throw new Error('{wasm contract} error that typically occurs when batch size is too large and node gas query limits are exceeded. Consider reducing the batch size.');
-    } else {
-      throw new Error(err);
-    }
-  }),
-);
+  client: SecretNetworkClient,
+  minBlockHeightValidationOptions?: MinBlockHeightValidationOptions,
+}) => {
+  let retryCount = 0;
+  return of(1).pipe( // placeholder observable of(1) used here so that we can start a data stream
+    // and retry from this level when certain error conditions are reached
+    switchMap(() => sendSecretClientContractQuery$({
+      queryMsg: msgBatchQuery(queries),
+      client,
+      contractAddress,
+      codeHash,
+    }).pipe(
+      map((response) => response as BatchQueryResponse), // map used for typecast only
+      switchMap((response) => {
+      // create an error if stale node is detected
+        if (minBlockHeightValidationOptions
+          && response.batch.block_height < minBlockHeightValidationOptions.minBlockHeight
+        ) {
+        // callback for when stale node is detected. Useful for error logging.
+          if (typeof minBlockHeightValidationOptions.onStaleNodeDetected === 'function') {
```

Comment:
I'd prefer if this only happens on the last retry

Reply:
nope, we want to be able to notify anytime a stale node is detected, even if the system is still able to run via retries.

Reply:
will add a if retryCount === 0 to ensure the callback only fires once